bazel_codegen/lib/src/summaries/build_asset_uri_resolver.dart

```diff
@@ -23,13 +23,17 @@ class BuildAssetUriResolver extends UriResolver {
     for (var asset in assets.where((asset) => asset.path.endsWith('.dart'))) {
       var uri = assetUri(asset);
       if (!_knownAssets.containsKey(uri)) {
-        _knownAssets[uri] = AssetSource(asset, await read(asset));
+        _knownAssets[uri] = AssetSource(asset, await read(asset), true);
       }
     }
   }
 
   @override
-  Source resolveAbsolute(Uri uri, [Uri actualUri]) => _knownAssets[uri];
+  Source resolveAbsolute(Uri uri, [Uri actualUri]) => _knownAssets.putIfAbsent(
```

Comment:
I worry about this returning empty sources for URIs that should be handled by summaries.

This resolver comes [earlier in the list](https://github.com/dart-lang/build/blob/fc63e34c54e2194b00d7c8d524ee1bd71d13a0b3/bazel_codegen/lib/src/summaries/analysis_context.dart#L32-L35) so I'm confused how that isn't already a problem...

Reply:
Good catch - I needed to move this to the end of the list.

Reply:
The api for that looks a lot different... @scheglov do you know what we should be doing for analysis driver?

Do we need to use `resolver.newFile` with null content or will it handle this better by default?